caom2-search-server/src/main/java/ca/nrc/cadc/search/SearchPreviewServlet.java

```diff
@@ -71,13 +76,39 @@ public SearchPreviewServlet(final PreviewRequestHandler previewRequestHandler) {
      * location.
      */
     public SearchPreviewServlet() {
+        this.profiler = new Profiler(SearchPreviewServlet.class);
+    }
+
+    @Override
+    protected void doGet(final HttpServletRequest req, final HttpServletResponse resp) throws IOException {
         final RegistryClient registryClient = new RegistryClient();
-        final URL dataServiceURL = registryClient.getServiceURL(
-            getServiceID(
-                CAOM2LINK_SERVICE_URI_PROPERTY_KEY,
-                DEFAULT_CAOM2LINK_SERVICE_URI),
-            Standards.DATALINK_LINKS_10, AuthMethod.COOKIE);
-        this.previewRequestHandler = new PreviewRequestHandler(dataServiceURL, new JobURLCreator() {
+        final String uriStr = req.getParameter("id");
+        URL serviceURL = null;
+        profiler.checkpoint(String.format("%s doGet() start", uriStr));
+
+        // PublisherID format expected: ivo://<resourceID>?<query string>
```

Comment:
the scheme is part of the resourceID so this comment is wrong: <resourceID>?<query string>